packages/nextjs/src/app-router/server/controlComponents.tsx

```diff
@@ -16,37 +15,51 @@ export function SignedOut(props: React.PropsWithChildren) {
   return userId ? null : <>{children}</>;
 }
 
-type GateServerComponentProps = React.PropsWithChildren<
-  Parameters<experimental__CheckAuthorizationWithoutPermission>[0] & {
-    fallback?: React.ReactNode;
-    redirectTo?: string;
-  }
->;
+type ProtectServerComponentProps = React.ComponentProps<typeof ProtectClientComponent>;
 
 /**
- * @experimental The component is experimental and subject to change in future releases.
+ * Use `<Protect/>` in order to prevent unauthenticated or unauthorized users from accessing the children passed to the component.
+ *
+ * Examples:
+ * ```
+ * <Protect permission="a_permission_key" />
+ * <Protect role="a_role_key" />
+ * <Protect condition={(has) => has({permission:"a_permission_key"})} />
+ * <Protect condition={(has) => has({role:"a_role_key"})} />
+ * <Protect fallback={<p>Unauthorized</p>} />
+ * ```
  */
-export function experimental__Gate(gateProps: GateServerComponentProps) {
-  const { children, fallback, redirectTo, ...restAuthorizedParams } = gateProps;
-  const { experimental__has } = auth();
-
-  const isAuthorizedUser = experimental__has(restAuthorizedParams);
+export function Protect(props: ProtectServerComponentProps) {
+  const { children, fallback, ...restAuthorizedParams } = props;
+  const { has, userId } = auth();
 
-  const handleFallback = () => {
-    if (!redirectTo && !fallback) {
-      throw new Error('Provide `<Gate />` with a `fallback` or `redirectTo`');
+  /**
+   * If neither of the authorization params are passed behave as the `<SignedIn/>`.
+   * If fallback is present render that instead of rendering nothing.
+   */
+  if (!restAuthorizedParams.condition && !restAuthorizedParams.role && !restAuthorizedParams.permission) {
```

Comment:
@panteliselef If the user is signed out, there is no point in checking any authorization parameters. We should exit early. So, all the conditional log in the following can be simplified with less nesting and more guard conditionals.

```
const f = <>fallback ?? null</>;

if (!userId) { 
  return f;
}

if (restParams.condition  === function) {
  ... do the function check
} 

if (restParams.roles || restParams.permissions) {
  ... do the has check
} 

return <>{children}</>
```